app/models/admin_user.rb

```diff
@@ -23,11 +27,54 @@ class AdminUser < ActiveRecord::Base
   validates_format_of :password, with: PASSWORD_REGEX, allow_blank: true
   validates_inclusion_of :role, in: ROLES
 
+  # = Callbacks =
+  before_update if: :crypted_password_changed? do
+    self.force_password_reset = false
+    self.password_changed_at = Time.current
+  end
+
   # = Finders =
   scope :by_name, -> { order(:last_name, :first_name) }
   scope :by_role, ->(role) { where(role: role).order(:id) }
 
   # = Methods =
+  def current_password
+    defined?(@current_password) ? @current_password : nil
+  end
+
+  def current_password=(value)
+    @current_password = value
+  end
+
+  def update_with_password(attrs)
```

Comment:
Similar comment to the below. I'm a fan of thin models used as a data access layer with business logic held elsewhere in a structured manner so I'd be looking to extract this outside the model.

However, if it's not a pattern present already in this project, then I guess it could be kept here, or moved to a concern.